orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -869,89 +869,216 @@ common::Status HandleSharedInitializer(Graph& graph,
   return Status::OK();
 }
 
-// split the graph into disconnected subgraph based on provided CutInfo
-common::Status SplitGraph(Graph& graph,
-                          std::vector<TrainingSession::TrainingConfiguration::CutInfo> split_edge_groups,
-                          std::vector<Node*>& send_nodes,
-                          std::vector<Node*>& recv_nodes) {
+// Returns all the pointers to NodeArg in the graph, before applying any 
+// partition transformation. 
+std::vector<const NodeArg*> GetAllNodeArgs(Graph& graph) {
+  std::vector<const NodeArg*> initial_node_args;
+  for (size_t i = 0, t = graph.MaxNodeIndex(); i < t; ++i) {
+    Node* node = graph.GetNode(i);
+    auto& node_outputs = node->MutableOutputDefs();
+    for (NodeArg* arg : node_outputs) {
+      if (arg == nullptr || !arg->HasTensorOrScalarShape()) 
+        continue;
+      initial_node_args.push_back(arg);
+    }
+  }
+  return initial_node_args;
+}
+
+common::Status AddMetaTensors(int current_rank, int next_rank,
```

Comment:
Suggest not doing this refactor here, as it moves the diffs around. -- try to re-inline AddMetaTensors below and see if the diff gets smaller?